AppDelegate.mm

```diff
@@ -96,6 +131,14 @@ - (void)applicationDidFinishLaunching:(NSNotification*)notification {
   NSDictionary* downloadURLs = [info objectForKey:@"TargetDownloadURLs"];
   NSURL* downloadURL = [NSURL URLWithString:[downloadURLs valueForKey:ARCH_KEY_NAME]];
   NSString* targetAppName = [info valueForKey:@"TargetAppName"];
+  auto* tempDir = [NSBundle.mainBundle.bundlePath stringByAppendingPathComponent:@"download"];
+  auto* fileManager = NSFileManager.defaultManager;
+
+  // Will be true if installer has failed before
+  BOOL folderExists = [fileManager fileExistsAtPath:tempDir];
+  if (folderExists) {
+    showErrorModal(@"Looks like your last installation did not complete.");
```

Comment:
This is not actionable for the user. We should provide them with something more useful like a button to download the platform-specific .zip manually.

Reply:
`showErrorModal` does exactly that. It provides a download button which opens the download URL.

Reply:
Oh lol, I forgot I added that 😅